code/extensions/RealMeSecurityExtension.php

```diff
@@ -91,17 +94,50 @@ public function realme()
      */
     private function realMeACS()
     {
-        $loggedIn = $this->service->enforceLogin();
+        try {
+            $authenticated = $this->service->enforceLogin();
+
+            if($authenticated === true) {
+                $authData = $this->service->getAuthData();
+
+                // If more session vars are set here, they must be cleared in realmeLogout()
+                Session::set('RealMeSessionData', serialize($authData));
+                Session::set('RealMeOriginalResponse', $_POST['SAMLResponse']);
```

Comment:
That may be suitable, I'm just very weary of storing raw external data in sessions. Most users assume session data is safe